src/compiler/tsc.ts

```diff
@@ -6,9 +6,19 @@ namespace ts {
         fileWatcher?: FileWatcher;
     }
 
-    let reportDiagnostic = reportDiagnosticSimply;
+    const defaultFormatDiagnosticsHost: FormatDiagnosticsHost = {
+        getCurrentDirectory: () => sys.getCurrentDirectory(),
+        getNewLine: () => sys.newLine,
+        getCanonicalFileName: createGetCanonicalFileName(sys.useCaseSensitiveFileNames)
+    };
+
+    let reportDiagnosticWorker = reportDiagnosticSimply;

```

Comment:
because `reportDiagnosticWorker` can be either `reportDiagnosticSimply` or `reportDiagnosticWithColorAndContext` if `pretty` option is set